File: backend/scripts/seed-firebase.py
"""
seed-firebase.py
────────────────
Seeds Firestore with:
  • 5 sample doctors  (doctors/ collection + users/ collection)
  • 1 admin user      (users/ collection)
  • 3 verified_doctors entries (for doctor sign-up license check)

Requirements:
    pip install firebase-admin

Setup:
    1. Go to Firebase Console → Project Settings → Service Accounts
    2. Click "Generate new private key" → save as serviceAccountKey.json
    3. Place serviceAccountKey.json in the same folder as this script
    4. Run:  python seed-firebase.py
"""

import firebase_admin
from firebase_admin import credentials, firestore
from datetime import datetime, timezone
import sys
import os

# ─── Config ──────────────────────────────────────────────────────────────────
SERVICE_ACCOUNT_PATH = os.path.join(os.path.dirname(__file__), "serviceAccountKey.json")

def init_firebase():
    if not os.path.exists(SERVICE_ACCOUNT_PATH):
        print("❌  serviceAccountKey.json not found!")
        print("    Download it from Firebase Console → Project Settings → Service Accounts")
        print(f"    Place it at: {SERVICE_ACCOUNT_PATH}")
        sys.exit(1)
    cred = credentials.Certificate(SERVICE_ACCOUNT_PATH)
    firebase_admin.initialize_app(cred)
    print("✅  Firebase initialized")

# ─── Doctors Seed Data ────────────────────────────────────────────────────────
DOCTORS = [
    {
        "uid": "doctor_seed_001",
        "email": "dr.rajesh.kumar@healthvault.com",
        "fullName": "Dr. Rajesh Kumar",
        "role": "doctor",
        "emailVerified": True,
        "accountStatus": "active",
        "profile": {
            "fullName": "Dr. Rajesh Kumar",
            "specialization": "Cardiology",
            "experience": "12 years",
            "consultationFee": "800",
            "licenseNumber": "MCI-2012-001",
            "phoneNumber": "+91-9876543210",
            "bio": "Senior Cardiologist specializing in interventional cardiology and heart failure management.",
            "availability": "Mon–Fri, 10:00 AM – 5:00 PM",
            "hospital": "City Heart Hospital",
        },
    },
    {
        "uid": "doctor_seed_002",
        "email": "dr.priya.sharma@healthvault.com",
        "fullName": "Dr. Priya Sharma",
        "role": "doctor",
        "emailVerified": True,
        "accountStatus": "active",
        "profile": {
            "fullName": "Dr. Priya Sharma",
            "specialization": "Dermatology",
            "experience": "8 years",
            "consultationFee": "600",
            "licenseNumber": "MCI-2016-002",
            "phoneNumber": "+91-9876543211",
            "bio": "Expert dermatologist focused on skin allergies, acne treatment, and cosmetic dermatology.",
            "availability": "Tue–Sat, 11:00 AM – 6:00 PM",
            "hospital": "SkinCare Clinic",
        },
    },
    {
        "uid": "doctor_seed_003",
        "email": "dr.amit.patel@healthvault.com",
        "fullName": "Dr. Amit Patel",
        "role": "doctor",
        "emailVerified": True,
        "accountStatus": "active",
        "profile": {
            "fullName": "Dr. Amit Patel",
            "specialization": "Orthopedics",
            "experience": "15 years",
            "consultationFee": "1000",
            "licenseNumber": "MCI-2009-003",
            "phoneNumber": "+91-9876543212",
            "bio": "Orthopedic surgeon specializing in joint replacement, sports injuries, and spine disorders.",
            "availability": "Mon–Fri, 9:00 AM – 4:00 PM",
            "hospital": "Bone & Joint Institute",
        },
    },
    {
        "uid": "doctor_seed_004",
        "email": "dr.sunita.mehta@healthvault.com",
        "fullName": "Dr. Sunita Mehta",
        "role": "doctor",
        "emailVerified": True,
        "accountStatus": "active",
        "profile": {
            "fullName": "Dr. Sunita Mehta",
            "specialization": "Pediatrics",
            "experience": "10 years",
            "consultationFee": "500",
            "licenseNumber": "MCI-2014-004",
            "phoneNumber": "+91-9876543213",
            "bio": "Dedicated pediatrician caring for children from newborns to adolescents.",
            "availability": "Mon–Sat, 10:00 AM – 2:00 PM",
            "hospital": "Children's Care Center",
        },
    },
    {
        "uid": "doctor_seed_005",
        "email": "dr.vikram.singh@healthvault.com",
        "fullName": "Dr. Vikram Singh",
        "role": "doctor",
        "emailVerified": True,
        "accountStatus": "active",
        "profile": {
            "fullName": "Dr. Vikram Singh",
            "specialization": "Neurology",
            "experience": "18 years",
            "consultationFee": "1200",
            "licenseNumber": "MCI-2006-005",
            "phoneNumber": "+91-9876543214",
            "bio": "Senior neurologist treating migraines, epilepsy, stroke, and neurodegenerative conditions.",
            "availability": "Wed–Sun, 12:00 PM – 7:00 PM",
            "hospital": "Brain & Spine Hospital",
        },
    },
]

# ─── Verified Doctors (for license check during sign-up) ─────────────────────
VERIFIED_DOCTORS = [
    {"licenseNumber": "MCI-2012-001", "fullName": "Dr. Rajesh Kumar",  "medicalCouncil": "Medical Council of India"},
    {"licenseNumber": "MCI-2016-002", "fullName": "Dr. Priya Sharma",  "medicalCouncil": "Medical Council of India"},
    {"licenseNumber": "MCI-2009-003", "fullName": "Dr. Amit Patel",    "medicalCouncil": "Medical Council of India"},
    {"licenseNumber": "MCI-2014-004", "fullName": "Dr. Sunita Mehta",  "medicalCouncil": "Medical Council of India"},
    {"licenseNumber": "MCI-2006-005", "fullName": "Dr. Vikram Singh",  "medicalCouncil": "Medical Council of India"},
]

# ─── Verified Admins ─────────────────────────────────────────────────────────
VERIFIED_ADMINS = [
    {"adminId": "ADM-001", "fullName": "System Admin",      "department": "Administration"},
    {"adminId": "ADM-002", "fullName": "Hospital Manager",  "department": "Management"},
    {"adminId": "ADM-003", "fullName": "Clinic Admin",      "department": "Operations"},
]

# ─── Admin User ───────────────────────────────────────────────────────────────
ADMIN_USER = {
    "uid": "admin_seed_001",
    "email": "admin@healthvault.com",
    "role": "admin",
    "emailVerified": True,
    "accountStatus": "active",
    "profile": {
        "fullName": "System Admin",
        "phoneNumber": "+91-9000000000",
    },
}

# ─── Seeder Functions ─────────────────────────────────────────────────────────
def seed_doctors(db):
    print("\n📋  Seeding doctors...")
    for doctor in DOCTORS:
        uid = doctor["uid"]
        now = datetime.now(timezone.utc)

        # Write to doctors/ collection (used for patient search)
        doctors_doc = {
            "uid": uid,
            "email": doctor["email"],
            "fullName": doctor["fullName"],
            "emailVerified": doctor["emailVerified"],
            "accountStatus": doctor["accountStatus"],
            "profile": doctor["profile"],
            "createdAt": now,
            "lastLogin": now,
        }
        db.collection("doctors").document(uid).set(doctors_doc)

        # Write to users/ collection (for role-based routing)
        users_doc = {
            "uid": uid,
            "email": doctor["email"],
            "role": doctor["role"],
            "emailVerified": doctor["emailVerified"],
            "accountStatus": doctor["accountStatus"],
            "createdAt": now,
            "lastLogin": now,
        }
        db.collection("users").document(uid).set(users_doc)

        print(f"   ✅  {doctor['fullName']} ({doctor['profile']['specialization']})")

    print(f"   → {len(DOCTORS)} doctors seeded")


def seed_verified_doctors(db):
    print("\n🏥  Seeding verified_doctors (license registry)...")
    for vd in VERIFIED_DOCTORS:
        doc_id = vd["licenseNumber"].replace("-", "_")
        db.collection("verified_doctors").document(doc_id).set(vd)
        print(f"   ✅  {vd['licenseNumber']} — {vd['fullName']}")
    print(f"   → {len(VERIFIED_DOCTORS)} license entries seeded")


def seed_verified_admins(db):
    print("\n🔐  Seeding verified_admins (admin ID registry)...")
    for adm in VERIFIED_ADMINS:
        doc_id = adm["adminId"].replace("-", "_")
        db.collection("verified_admins").document(doc_id).set(adm)
        print(f"   ✅  {adm['adminId']} — {adm['fullName']}")
    print(f"   → {len(VERIFIED_ADMINS)} admin IDs seeded")


def seed_admin(db):
    print("\n👤  Seeding admin user...")
    now = datetime.now(timezone.utc)
    db.collection("users").document(ADMIN_USER["uid"]).set({
        **ADMIN_USER,
        "createdAt": now,
        "lastLogin": now,
    })
    print(f"   ✅  Admin: {ADMIN_USER['email']}")
    print("   ⚠️  NOTE: Create this user in Firebase Auth manually with")
    print(f"       email: {ADMIN_USER['email']}  password: Admin@123")
    print("       Then update the UID in this script to match the Firebase Auth UID")


def main():
    print("=" * 55)
    print("  Smart Health Vault — Firebase Seeder")
    print("=" * 55)

    init_firebase()
    db = firestore.client()

    seed_verified_doctors(db)
    seed_verified_admins(db)
    seed_doctors(db)
    seed_admin(db)

    print("\n" + "=" * 55)
    print("✅  Seeding complete!")
    print("=" * 55)
    print("\nNext steps:")
    print("  1. Deploy indexes:  firebase deploy --only firestore:indexes")
    print("  2. Deploy rules:    firebase deploy --only firestore:rules")
    print("  3. Run the app and sign up a doctor using one of these")
    print("     license numbers to test the doctor sign-up flow:")
    for vd in VERIFIED_DOCTORS:
        print(f"     • {vd['licenseNumber']}  ({vd['fullName']})")

if __name__ == "__main__":
    main()

File: backend/scripts/seed-firebase.js
const admin = require('firebase-admin');
const path = require('path');

// Initialize Firebase Admin
const serviceAccount = require('../firebase-service-account.json');

admin.initializeApp({
    credential: admin.credential.cert(serviceAccount),
});

const db = admin.firestore();

// ============================================================================
// TEST STAFF ACCOUNT - CHANGE THIS TO YOUR EMAIL
// ============================================================================

// 👇 CHANGE THIS TO YOUR EMAIL ADDRESS
const YOUR_EMAIL = 'dhruvamtandel0412@gmail.com'; // Your email for admin staff account


const TEST_STAFF = {
    email: YOUR_EMAIL,
    organizationId: 'ORG001',
    employeeId: 'EMP001',
    organizationName: 'City Hospital',
    department: 'Administration',
    fullName: 'Admin Staff',
    phone: '+1234567890',
};

// ============================================================================
// VERIFIED ORGANIZATIONS
// ============================================================================

const VERIFIED_ORGANIZATIONS = [
    {
        id: 'ORG001',
        name: 'City Hospital',
        address: '123 Main Street, New York, NY 10001',
        phone: '+1-555-0101',
        email: 'contact@cityhospital.com',
        type: 'Hospital',
        verified: true,
        emailDomains: ['cityhospital.com', 'gmail.com'], // Added gmail.com for testing
        createdAt: new Date().toISOString(),
    },
    {
        id: 'ORG002',
        name: 'Metro Medical Center',
        address: '456 Park Avenue, Los Angeles, CA 90001',
        phone: '+1-555-0102',
        email: 'info@metromedical.com',
        type: 'Medical Center',
        verified: true,
        emailDomains: ['metromedical.com', 'gmail.com'], // Added gmail.com for testing
        createdAt: new Date().toISOString(),
    },
    {
        id: 'ORG003',
        name: 'Downtown Clinic',
        address: '789 Oak Street, Chicago, IL 60601',
        phone: '+1-555-0103',
        email: 'contact@downtownclinic.com',
        type: 'Clinic',
        verified: true,
        emailDomains: ['downtownclinic.com', 'gmail.com'], // Added gmail.com for testing
        createdAt: new Date().toISOString(),
    },
];


// ============================================================================
// VERIFIED DOCTORS
// ============================================================================

const VERIFIED_DOCTORS = [
    {
        licenseNumber: 'MH12345',
        fullName: 'Dr. John Smith',
        specialization: 'Cardiology',
        medicalCouncil: 'Maharashtra Medical Council',
        qualification: 'MBBS, MD (Cardiology)',
        verified: true,
        createdAt: new Date().toISOString(),
    },
    {
        licenseNumber: 'DL67890',
        fullName: 'Dr. Sarah Johnson',
        specialization: 'Pediatrics',
        medicalCouncil: 'Delhi Medical Council',
        qualification: 'MBBS, MD (Pediatrics)',
        verified: true,
        createdAt: new Date().toISOString(),
    },
    {
        licenseNumber: 'KA11223',
        fullName: 'Dr. Michael Brown',
        specialization: 'Orthopedics',
        medicalCouncil: 'Karnataka Medical Council',
        qualification: 'MBBS, MS (Orthopedics)',
        verified: true,
        createdAt: new Date().toISOString(),
    },
    {
        licenseNumber: 'TN44556',
        fullName: 'Dr. Emily Davis',
        specialization: 'Dermatology',
        medicalCouncil: 'Tamil Nadu Medical Council',
        qualification: 'MBBS, MD (Dermatology)',
        verified: true,
        createdAt: new Date().toISOString(),
    },
    {
        licenseNumber: 'GJ77889',
        fullName: 'Dr. Robert Wilson',
        specialization: 'Neurology',
        medicalCouncil: 'Gujarat Medical Council',
        qualification: 'MBBS, DM (Neurology)',
        verified: true,
        createdAt: new Date().toISOString(),
    },
];

// ============================================================================
// VERIFIED STAFF MEMBERS
// ============================================================================

const VERIFIED_STAFF = [
    {
        organizationId: 'ORG001',
        employeeId: 'EMP001',
        email: YOUR_EMAIL, // Uses your custom email
        fullName: 'Admin Staff',
        department: 'Administration',
        position: 'Admin',
        verified: true,
        createdAt: new Date().toISOString(),
    },
    {
        organizationId: 'ORG001',
        employeeId: 'EMP002',
        email: 'nurse.jane@cityhospital.com',
        fullName: 'Jane Nurse',
        department: 'Nursing',
        position: 'Senior Nurse',
        verified: true,
        createdAt: new Date().toISOString(),
    },
    {
        organizationId: 'ORG002',
        employeeId: 'EMP003',
        email: 'admin.metro@metromedical.com',
        fullName: 'Metro Admin',
        department: 'Administration',
        position: 'Admin Officer',
        verified: true,
        createdAt: new Date().toISOString(),
    },
];

// ============================================================================
// SEED FUNCTIONS
// ============================================================================

async function seedVerifiedOrganizations() {
    console.log('\n📋 Seeding verified organizations...');

    for (const org of VERIFIED_ORGANIZATIONS) {
        try {
            await db.collection('verified_organizations').doc(org.id).set(org);
            console.log(`✅ Added organization: ${org.name} (${org.id})`);
        } catch (error) {
            console.error(`❌ Error adding organization ${org.id}:`, error.message);
        }
    }
}

async function seedVerifiedDoctors() {
    console.log('\n👨‍⚕️ Seeding verified doctors...');

    for (const doctor of VERIFIED_DOCTORS) {
        try {
            await db.collection('verified_doctors').doc(doctor.licenseNumber).set(doctor);
            console.log(`✅ Added doctor: ${doctor.fullName} (${doctor.licenseNumber})`);
        } catch (error) {
            console.error(`❌ Error adding doctor ${doctor.licenseNumber}:`, error.message);
        }
    }
}

async function seedVerifiedStaff() {
    console.log('\n👔 Seeding verified staff members...');

    for (const staff of VERIFIED_STAFF) {
        try {
            const docId = `${staff.organizationId}_${staff.employeeId}`;
            await db.collection('verified_staff').doc(docId).set(staff);
            console.log(`✅ Added staff: ${staff.fullName} (${staff.organizationId}/${staff.employeeId})`);
        } catch (error) {
            console.error(`❌ Error adding staff ${staff.employeeId}:`, error.message);
        }
    }
}

// ============================================================================
// MAIN FUNCTION
// ============================================================================

async function seedAll() {
    console.log('🌱 Starting Firebase seeding process...\n');
    console.log('='.repeat(60));

    try {
        await seedVerifiedOrganizations();
        await seedVerifiedDoctors();
        await seedVerifiedStaff();

        console.log('\n' + '='.repeat(60));
        console.log('\n✅ Seeding completed successfully!');
        console.log('\n📊 Summary:');
        console.log(`   - Organizations: ${VERIFIED_ORGANIZATIONS.length}`);
        console.log(`   - Doctors: ${VERIFIED_DOCTORS.length}`);
        console.log(`   - Staff Members: ${VERIFIED_STAFF.length}`);

        console.log('\n🧪 Test Credentials:');
        console.log('\n   📧 Test Staff Account:');
        console.log(`      Email: ${TEST_STAFF.email}`);
        console.log(`      Organization ID: ${TEST_STAFF.organizationId}`);
        console.log(`      Employee ID: ${TEST_STAFF.employeeId}`);
        console.log(`      Password: (Set during signup)`);

        console.log('\n   👨‍⚕️ Test Doctor Licenses:');
        VERIFIED_DOCTORS.forEach(doc => {
            console.log(`      - ${doc.licenseNumber} (${doc.fullName})`);
        });

        console.log('\n   🏥 Test Organizations:');
        VERIFIED_ORGANIZATIONS.forEach(org => {
            console.log(`      - ${org.id} (${org.name})`);
        });

        console.log('\n' + '='.repeat(60));

    } catch (error) {
        console.error('\n❌ Seeding failed:', error);
        process.exit(1);
    }

    process.exit(0);
}

// Run seeding
seedAll();
